src/main/java/sb/web/OrderController.java

```diff
@@ -0,0 +1,86 @@
+package sb.web;
+
+import org.springframework.data.repository.query.Param;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import sb.domain.dto.request.OrderDTORequest;
+import sb.domain.entity.Order;
+import sb.domain.entity.User;
+import sb.utils.mapper.OrderMapper;
+import sb.domain.dto.response.OrderDTOResponse;
+import sb.service.OrderService;
+import sb.service.UserService;
+import sb.service.exception.CreationException;
+import sb.utils.OrderSort;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+
+    private OrderService orderService;
+    private UserService userService;
+    private OrderMapper orderMapper;
+    private OrderSort orderSort;
+
+    public OrderController(OrderService orderService, UserService userService, OrderMapper orderMapper, OrderSort orderSort) {
+        this.orderService = orderService;
+        this.userService = userService;
+        this.orderMapper = orderMapper;
+        this.orderSort = orderSort;
+    }
+
+    @GetMapping
+    public List<OrderDTOResponse> getAll(@Param("status") String status, @Param("sortBy") String sortBy) {
+        List<OrderDTOResponse> orderDTOs = orderMapper.allToModel(orderService.getAll());
+        orderDTOs = orderSort.sort(sortBy, status, orderDTOs);
+        return orderDTOs;
+    }
+
+    @GetMapping("/{id}")
+    public OrderDTOResponse get(@PathVariable("id") int id) {
+        return orderMapper.toModel(orderService.get(id));
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public OrderDTOResponse create(@RequestBody @Valid OrderDTORequest orderDTORequest, BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            throw new CreationException(bindingResult);
```

Comment:
why don't you handle exceptions in exception handler? including the ones related to binding. Looks like SRP is violated here

Reply:
fix: remove extraargument, handle validationException it in Handler